### PR TITLE
Enabling test reporting for all tests

### PR DESCRIPTION
**Pull request by kiiadi**
## Description
Enabling CodeBuild [Test Reports](https://aws.amazon.com/blogs/devops/test-reports-with-aws-codebuild/)

## Motivation and Context
It's annoying having to scroll through 1000s of lines of logs to figure out which tests fail. This also gives us the ability to track tests over time.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/testRunner.ts [hunk 1]
```diff
@@ -11,10 +11,7 @@ export interface RunTestsParameters {
 
 export async function runTests(parameters: RunTestsParameters): Promise<void> {
     if (process.env.AWS_TOOLKIT_TEST_NO_COLOR) {
-        legacyTestRunner.configureMocha({
-            ui: 'bdd',
-            useColors: false
-        })
+        legacyTestRunner.configureMocha(legacyTestRunner.defaultMochaOptions(false))
```
Comment (awschristou) [inline src/test/testRunner.ts]: Thank you, this is a nice change that re-internalizes the defaults

File: third-party/test/index.ts [hunk 2]
```diff
@@ -38,6 +38,7 @@ import * as glob from "glob";
 import * as os from "os"
 // END 2018-10-05: Amazon addition.
 import * as paths from "path";
+import { MochaOptions } from "mocha";
```
Comment (awschristou) [inline third-party/test/index.ts]: All changes to this third-party file need to be divergence commented (see examples within the file)

Comment (kiiadi) [reply third-party/test/index.ts]: i dig

File: package.json [hunk 3]
```diff
@@ -507,6 +507,7 @@
         "lolex": "^4.2.0",
         "marked": "^0.7.0",
         "mocha": "^6.2.0",
+        "mocha-junit-reporter": "^1.23.3",
```
Comment (awschristou) [inline package.json]: did you add this by hand or with npm? I think you'll need to use npm to do this in order to address the build issues.

Comment (kiiadi) [reply package.json]: i did some npm command... the only build issues i'm seeing are on windows...

File: third-party/test/index.ts [hunk 4]
```diff
@@ -52,10 +56,23 @@ if (!tty.getWindowSize) {
...
 
-let mocha = new Mocha({
-    ui: "bdd",
-    useColors: true,
-});
+// 2020-02-05: Amazon addition.
+export function defaultMochaOptions(useColors: boolean = true): MochaOptions {
+    return {
+        ui: "bdd",
+        useColors: useColors,
```
Comment (bryceitoc9) [inline third-party/test/index.ts]: Slight style nit: `useColors: useColors,` can be shortened to `useColors,` Up to you if you want to change it.

**Pull request merged**